pipeline/cleanup/cleanup.go

```diff
@@ -0,0 +1,42 @@
+package cleanup
+
+import (
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"strings"
+
+	"github.com/goreleaser/goreleaser/context"
+)
+
+type Pipe struct{}
+
+func (Pipe) String() string {
+	return "Prepare release"
+}
+
+func (Pipe) Default(ctx *context.Context) error {
+	if ctx.Config.Cleanup.Hooks == nil {
+		ctx.Config.Cleanup.Hooks = []string{}
+	}
+	return nil
+}
+
+func (Pipe) Run(ctx *context.Context) error {
+	for _, step := range ctx.Config.Cleanup.Hooks {
+		args := strings.Fields(step)
+		cmd := exec.Command(args[0], args[1:]...)
+		if out, err := cmd.CombinedOutput(); err != nil {
+			return fmt.Errorf("hook failed: %s\n%v", step, string(out))
+		}
+	}
+	dir, err := filepath.Abs(ctx.Config.Dist)
```

Comment:
the dist cleanup is already done with the `--rm-dist` flag...

but I liked the "sneaky dir" check, maybe we could add this to the `dist` pipe?

Reply:
See https://github.com/goreleaser/goreleaser/pull/623